### ISSUE TITLE
Match Syntax: Multiple not-words

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#997]
The match-syntax `!foo{0,5}` works well for matching: _any word (except for the word `foo`) 0-5 times_
Is there any match syntax for expanding this to multiple not-want words?

I have tried:
- `!(foo|bar){0,5}`
- `(!foo|!bar){0,5}`
- `(!foo && !bar){0,5}`

Here's some playing around:

```javascript

const nlp = require('compromise')
const patterns = ['before (!foo|!bar){0,8}', 'before (!foo && !bar){0,8} something', 'before !(foo|bar){0,8} something']

const sentences = [
  { want: true, text: 'before one two three four five six seven something' },
  { want: false, text: 'before one bar two three something' },
  { want: false, text: 'before one two foo four five six something' },
]

patterns.forEach((pattern) => {
  console.log(pattern)
  sentences.forEach((sentence) => {
    const result = nlp(sentence.text).has(pattern) === sentence.want ? '✅' : '❌'
    console.log(`${result} ${sentence.text}`)
  })
  console.log()
})
```

## Output

I'm failing to get an output where all three sentences get ✅ for a pattern...

```ascii
before (!foo|!bar){0,8}
✅ before one two three four five six seven something
❌ before one bar two three something
❌ before one two foo four five six something

before (!foo && !bar){0,8} something
❌ before one two three four five six seven something
✅ before one bar two three something
✅ before one two foo four five six something

before !(foo|bar){0,8} something
✅ before one two three four five six seven something
❌ before one bar two three something
❌ before one two foo four five six something
```

**Comment by spencermountain** [comment]
yeah - `!(foo|bar){0,3}` should work - looks like the tokenizer bonks the syntax
```js
console.log(nlp.parseMatch('match !(foo|bar){0,3}'))
```
happy to take a look at this

**Comment by spencermountain** [comment]
fixed on dev

**Comment by spencermountain** [comment]
fixed in 14.9.0
thanks

**Comment by vjsnagglepuss** [comment]
thank you so much for this fix @spencermountain -- it will make my life much easier!
